pkg/apis/postgres-operator.crunchydata.com/v1beta1/instrumentation_types.go

```diff
@@ -65,6 +65,14 @@ type InstrumentationConfigSpec struct {
 	// +listType=atomic
 	// +optional
 	Files []corev1.VolumeProjection `json:"files,omitempty"`
+
+	// EnvironmentVariables allows the user to add environment variables to the
+	// collector container.
+	// ---
+	// +kubebuilder:validation:MinItems=1
+	// +listType=atomic
+	// +optional
+	EnvironmentVariables []corev1.EnvVar `json:"environmentVariables,omitempty"`
```

Comment:
🤔 This seems reasonable for the following reasons:

- there are many exporters, etc.
- we want to support many of them
- many of them require some env vars to be set, with different names
- therefore the comprehensive way to support as many exporters as possible is to allow the user to set any env var they want.

On the other hand, if we let people set any env var they want... is there some potential harm here? For one thing, do we want people putting passwords and keys into the spec? For another, is there a potential problem in, say, overriding one of the env vars that we need to be set in the collector container?

Reply:
I really want to specify an allowlist of envvar names people are allowed to use, but that runs into the issue of letting people choose their OTEL modules...